BASE/src/Microsoft.ApplicationInsights/Extensibility/TelemetryConfiguration.cs

```diff
@@ -399,6 +405,16 @@ public void Dispose()
             GC.SuppressFinalize(this);
         }
 
+        /// <summary>
+        /// Set a TokenCredential for this configuration.
+        /// </summary>
+        /// <param name="tokenCredential">An instance of Azure.Core.TokenCredential.</param>
+#if NETSTANDARD2_0
+        public void SetCredential(Azure.Core.TokenCredential tokenCredential) => this.CredentialEnvelope = new TokenCredentialEnvelope(tokenCredential);
+#else
+        public void SetCredential(object tokenCredential) => this.CredentialEnvelope = new ReflectionCredentialEnvelope(tokenCredential);
+#endif
```

Comment:
We have a problem with this. The profiler agent is a net472 executable so, when it references the Application Insights SDK, it picks up the net46 flavor (SetCredential types as object). However, the profiler agent also references another class library (for logging) that's built for netstandard2.0. _That_ class library references the ApplicationInsights SDK too but, since it's netstandard2.0, it picks up the netstandard2.0 flavor of App Insights SDK and, therefore, the TokenCredential flavor of this API. 

When it all comes together, there's a MissingMethodException at run time because the EXE loads the net46 version, but the logger tries to call the SetCredential(TokenCredential...) method which doesn't exist in net46.

The way it is now, in order to address this, we have to make our logging library multi-target net472 and netstandard2.0.

Reply:
Managing the conflicting frameworks has been the most complicated part of all of this! :)
I'm confused why your net472 agent isn't taking the netstandard2.0 version of our SDK.

@pharring **Would it help you if we target net462?**

As of today, this sdk targets net452, net46, and netstandard2.0.
[Official guidance](https://devblogs.microsoft.com/dotnet/net-framework-4-5-2-4-6-4-6-1-will-reach-end-of-support-on-april-26-2022/) is to migrate all projects off of 4.5.2, 4.6, 4.6.1 to 4.6.2 by April 26, 2022.

Reply:
We faced similar issues in OpenTelemetry as well.

@utpilla as well who investigated this, and following up with .NET team to get an official guidance on this, if any.

Reply:
> I'm confused why your net472 agent isn't taking the netstandard2.0 version of our SDK.

Since our project has a netstandard and netframework target, a net472 project "prefers" the netframework version. So it picks net46 as thats closes to net472, and not netstandard2.0

Reply:
Thanks for the clarification @cijothomas .
In my opinion, we should include support for 4.6.2 sooner rather than later. I can start a new branch to investigate how much work this will be.
We don't need to retire older frameworks until next year.

Reply:
@cijothomas answered correctly above.

@TimothyMothra, I think adding 4.6.2 support would help in the meantime.

I suspect, once you retire net45-net461, you'll probably land where I did with Snapshot Collector 1.4-pre : just ship a netstandard2.0 version (Covered [here](https://github.com/microsoft/ApplicationInsights-SnapshotCollector/discussions/5))

Reply:
Would we consider the following approach and what's the pros/cons?

Always use `SetCredential(object tokenCredential)` signature, and avoid any static dependency (package dependency and assembly reference) to the Azure SDK. Use runtime detection / reflection.

Reply:
^^^ that would work for us, I think.

Reply:
Done